firedrake/slate/slac/utils.py

```diff
@@ -137,6 +136,19 @@ def visit_Symbol(self, o, *args, **kwargs):
         return SymbolWithFuncallIndexing(o.symbol, o.rank, o.offset)
 
 
+def count_operands(expr):
+    """Counts the total number of nodes in a given Slate expression.
+
+    :arg expr: A Slate expression.
+    """
+    count = 1
+
+    for op in expr.operands:
+        count += count_operands(op)
+
+    return count
```

Comment:
This doesn't dagify, is that intentional?

Reply:
I didn't have DAGs in mind when I wrote this. This function is used to sort the expressions requiring auxiliary temporaries. I want to order them in such a way that I don't run into an instance where I'm declaring a temporary for an expression which *may* depend on another temporary that hasn't been declared yet.

I just learned today that what I actually want is to *topologically sort* the expressions. So I will replace this function is a top-sort algorithm.